lint/private/lint_aspect.bzl

```diff
@@ -46,33 +46,75 @@ def should_visit(rule, allow_kinds, allow_filegroup_tags = []):
+
     if ctx.attr._options[LintOptionsInfo].fail_on_violation:
         # Fail on violation means the exit code is reported to Bazel as the action result
-        exit_code = None
+        human_exit_code = None
+        machine_exit_code = None
     else:
-        # The exit code should instead be provided as an action output so the build succeeds.
+        # The exit codes should instead be provided as action outputs so the build succeeds.
         # Downstream tooling like `aspect lint` will be responsible for reading the exit codes
         # and interpreting them.
-        exit_code = ctx.actions.declare_file(_OUTFILE_FORMAT.format(label = target.label.name, mnemonic = mnemonic, suffix = "exit_code"))
-        outs.append(exit_code)
-    return report, exit_code, OutputGroupInfo(rules_lint_report = depset(outs))
+        human_exit_code = ctx.actions.declare_file(_OUTFILE_FORMAT.format(label = target.label.name, mnemonic = mnemonic, suffix = "txt.exit_code"))
+        machine_exit_code = ctx.actions.declare_file(_OUTFILE_FORMAT.format(label = target.label.name, mnemonic = mnemonic, suffix = "report.exit_code"))
+
+    human_outputs = [f for f in [human_out, human_exit_code] if f]
+    machine_outputs = [f for f in [machine_out, machine_exit_code] if f]
+    return struct(
+        human = struct(
+            stdout = human_out,
+            exit_code = human_exit_code,
+        ),
+        machine = struct(
+            stdout = machine_out,
+            exit_code = machine_exit_code,
+        ),
+    ), OutputGroupInfo(
+        rules_lint_human = depset(human_outputs),
+        rules_lint_machine = depset(machine_outputs),
+        # Legacy name used by existing callers.
+        # TODO(2.0): remove
+        rules_lint_report = depset(machine_outputs),
+    )
 
 def patch_file(mnemonic, target, ctx):
     patch = ctx.actions.declare_file(_OUTFILE_FORMAT.format(label = target.label.name, mnemonic = mnemonic, suffix = "patch"))
     return patch, OutputGroupInfo(rules_lint_patch = depset([patch]))
 
 # If we return multiple OutputGroupInfo from a rule implementation, only one will get used.
 # So we need a separate function to return both.
-def patch_and_report_files(*args):
+# buildifier: disable=function-docstring
+def patch_and_output_files(*args):
     patch, _ = patch_file(*args)
-    report, exit_code, _ = report_files(*args)
-    return patch, report, exit_code, OutputGroupInfo(
-        rules_lint_report = depset([f for f in [report, exit_code] if f]),
+    outputs, _ = output_files(*args)
+    human_outputs = [outputs.human.stdout, outputs.human.exit_code]
+    machine_outputs = [outputs.machine.stdout, outputs.machine.exit_code]
+    return struct(
+        human = outputs.human,
+        machine = outputs.machine,
+        patch = patch,
+    ), OutputGroupInfo(
+        rules_lint_human = depset(human_outputs),
```

Comment:
`rules_lint_report_human` && `rules_lint_report_machine` ??

Reply:
we decided it's okay like this